Source/Core/DolphinQt/Config/Mapping/IOWindow.cpp

```diff
@@ -51,18 +182,55 @@ void IOWindow::CreateMainLayout()
   m_option_list = new QListWidget();
   m_select_button = new QPushButton(tr("Select"));
   m_detect_button = new QPushButton(tr("Detect"));
-  m_or_button = new QPushButton(tr("| OR"));
-  m_and_button = new QPushButton(tr("&& AND"));
-  m_add_button = new QPushButton(tr("+ ADD"));
-  m_not_button = new QPushButton(tr("! NOT"));
   m_test_button = new QPushButton(tr("Test"));
-  m_expression_text = new QPlainTextEdit();
   m_button_box = new QDialogButtonBox();
   m_clear_button = new QPushButton(tr("Clear"));
   m_apply_button = new QPushButton(tr("Apply"));
   m_range_slider = new QSlider(Qt::Horizontal);
   m_range_spinbox = new QSpinBox();
 
+  m_parse_text = new QLineEdit();
+  m_parse_text->setReadOnly(true);
+
+  m_expression_text = new QPlainTextEdit();
+  m_expression_text->setFont(QFontDatabase::systemFont(QFontDatabase::FixedFont));
+  new ControlExpressionSyntaxHighlighter(m_expression_text->document(), m_parse_text);
+
+  m_operators_combo = new QComboBox();
+  m_operators_combo->addItem(tr("Operators"));
+  m_operators_combo->insertSeparator(1);
+  if (m_type == Type::Input)
+  {
+    m_operators_combo->addItem(tr("! Not"));
+    m_operators_combo->addItem(tr("* Multiply"));
+    m_operators_combo->addItem(tr("/ Divide"));
+    m_operators_combo->addItem(tr("% Modulo"));
+    m_operators_combo->addItem(tr("+ Add"));
+    m_operators_combo->addItem(tr("- Subtract"));
+    m_operators_combo->addItem(tr("> Greater-than"));
+    m_operators_combo->addItem(tr("< Less-than"));
+    m_operators_combo->addItem(tr("& And"));
+  }
+  m_operators_combo->addItem(tr("| Or"));
+  if (m_type == Type::Input)
+  {
+    m_operators_combo->addItem(tr(", Comma"));
+  }
+
+  m_functions_combo = new QComboBox();
+  m_functions_combo->addItem(tr("Functions"));
```

Comment:
Not needed in this PR but even with documentation available it might be helpful to have some sort of description.  Many of these are self explanatory but a couple aren't as obvious.

Reply:
We could have a wiki page with documentation on the functions.